sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/math.scala

```diff
@@ -258,6 +258,10 @@ case class Pow(left: Expression, right: Expression)
 
 case class Logarithm(left: Expression, right: Expression)
   extends BinaryMathExpression((c1, c2) => math.log(c2) / math.log(c1), "LOG") {
+
+  /**
+   * Natural log, i.e. using e as the base.
+   */
   def this(child: Expression) = {
     this(EulerNumber(), child)
   }
```

Comment:
I'd prefer to remove this single parameter constructor, because this would make both `ln` and `log(c)` express the same meaning in a SQL expression.

And also complex the binary log here to check its first argument branch in both `eval` & `genCode`